### PR TITLE
Feat: create two new endpoints to get all time and current week  data…

### PR DESCRIPTION
**Pull request by chethana-dissanayka**
… with player data  using  player id and user id

### CODE REVIEW

File: src/modules/kite-data/kite-data.service.ts [hunk 1]
```diff
@@ -148,73 +180,232 @@ export class KiteDataService {
...
+          user_id: "$player_details.user_id",
+          name: "$player_details.name",
+          city: "$player_details.city",
+          img_url: "$player_details.img_url",
+          kite_height: 1,
+          rank: 1
+        }
+      },
+      {
+        $match: {
```
Comment (lihini) [inline src/modules/kite-data/kite-data.service.ts:330]: Do this filtering stage at the top of the aggregation pipeline

File: src/modules/kite-data/kite-data.service.ts [hunk 2]
```diff
@@ -148,73 +180,232 @@ export class KiteDataService {
...
+
       let currentWeekData = {};
       if (includeCurrentWeek) {
         const [
           currentWeekTotalHeight,
           currentWeekTotalAttempts,
           currentWeekTotalFlyingMins,
           currentWeekMaxHeight,
-          currentWeekMinHeight
+          currentWeekMinHeight,
```
Comment (lihini) [inline src/modules/kite-data/kite-data.service.ts:360]: Remove this calculation

File: src/modules/kite-data/kite-data.service.ts [hunk 3]
```diff
@@ -266,16 +468,76 @@ export class KiteDataService {
...
     return await this.kiteDatumModel.aggregate(aggregationPipeline).exec();
   }
 
+
+
+  async getTotalPlayersCount() {
+    const aggregationPipeline: any[] = [
+      {
+          $match: {
+              "metadata.kite_player_id": { $exists: true }
```
Comment (lihini) [inline src/modules/kite-data/kite-data.service.ts]: Is this filtering stage necessary?

File: src/modules/kite-players/kite-players.module.ts [hunk 4]
```diff
@@ -22,6 +22,8 @@ import { KitePlayersService } from './kite-players.service';
   ],
   controllers: [KitePlayersController],
   providers: [KitePlayersService, KitePlayer],
-  exports: [KitePlayersService],
+  exports: [KitePlayersService,
```
Comment (lihini) [inline src/modules/kite-players/kite-players.module.ts]: Format this export to the line below

**Review by lihini — CHANGES_REQUESTED** [review]

File: src/modules/kite-data/kite-data.service.ts (hunk 1 above)
Comment (chethana-dissanayka) [reply src/modules/kite-data/kite-data.service.ts:330]: This filtering stage should be at the bottom of the aggregation pipeline to get rank.

File: src/modules/kite-data/kite-data.service.ts (hunk 2 above)
Comment (chethana-dissanayka) [reply src/modules/kite-data/kite-data.service.ts:360]: Done.

File: src/modules/kite-data/kite-data.service.ts (hunk 3 above)
Comment (chethana-dissanayka) [reply src/modules/kite-data/kite-data.service.ts]: Removed.

**Pull request merged**